[Quant][Eager] Added 4 bit support for eager mode quantization flow

dzdang opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #69846
* __->__ #69806
* #69769
* #69768

Summary: Minor modifications were made to support 4 bit embedding quantized module in eager mode quantization flow and to allow for testing of the changes

Reviewers: jerryzh168

Subscribers: jerryzh168, supriyar

Test plan: In pytorch main dir, execute
```
python test_quantization.py TestPostTrainingStatic.test_quantized_embedding
```
to run the series of tests, including the newly added test_embedding_4bit
function

Tasks: T106931792

Tags

Differential Revision: [D33152675](https://our.internmc.facebook.com/intern/diff/D33152675)

dzdang committed:
[Quant][Eager] Added 4 bit support for eager mode quantization flow

Summary: Minor modifications were made to support 4 bit embedding quantized module in eager mode quantization flow and to allow for testing of the changes

Reviewers: jerryzh168

Subscribers: jerryzh168, supriyar

Test plan: In pytorch main dir, execute
```
python test_quantization.py TestPostTrainingStatic.test_quantized_embedding
```
to run the series of tests, including the newly added test_embedding_4bit
function

Tasks: T106931792

Tags

[ghstack-poisoned]

dzdang committed:
Update on "[Quant][Eager] Added 4 bit support for eager mode quantization flow"


Summary: Minor modifications were made to support 4 bit embedding quantized module in eager mode quantization flow and to allow for testing of the changes

Reviewers: jerryzh168

Subscribers: jerryzh168, supriyar

Test plan: In pytorch main dir, execute
```
python test_quantization.py TestPostTrainingStatic.test_quantized_embedding
```
to run the series of tests, including the newly added test_embedding_4bit
function

Tasks: T106931792

Tags

[ghstack-poisoned]

dzdang committed:
Update on "[Quant][Eager] Added 4 bit support for eager mode quantization flow"


Summary: Minor modifications were made to support 4 bit embedding quantized module in eager mode quantization flow and to allow for testing of the changes

Reviewers: jerryzh168

Subscribers: jerryzh168, supriyar

Test plan: In pytorch main dir, execute
```
python test_quantization.py TestPostTrainingStatic.test_quantized_embedding
```
to run the series of tests, including the newly added test_embedding_4bit
function

Tasks: T106931792

Tags

[ghstack-poisoned]

dzdang committed:
Update on "[Quant][Eager] Added 4 bit support for eager mode quantization flow"


Summary: Minor modifications were made to support 4 bit embedding quantized module in eager mode quantization flow and to allow for testing of the changes

Reviewers: jerryzh168

Subscribers: jerryzh168, supriyar

Test plan: In pytorch main dir, execute
```
python test_quantization.py TestPostTrainingStatic.test_quantized_embedding
```
to run the series of tests, including the newly added test_embedding_4bit
function

Tasks: T106931792

Tags

[ghstack-poisoned]

jerryzh168 commented on torch/nn/quantized/modules/embedding_ops.py:
```diff
@@ -158,13 +158,12 @@ def from_float(cls, mod):
-        assert dtype == torch.quint8, \
-            f'The only supported weight dtype for nnq.Embedding is torch.quint8, got {dtype}'
+        assert dtype == torch.quint8 or dtype == torch.quint4x2, \
+            f'The only supported dtype for nnq.EmbeddingBag is torch.quint8 and torch.quint4x2, got {dtype}'
```
nit: is this nnq.Embedding?

jerryzh168 reviewed (approved):
Looks good

dzdang committed:
Update on "[Quant][Eager] Added 4 bit support for eager mode quantization flow"


Summary: Minor modifications were made to support 4 bit embedding quantized module in eager mode quantization flow and to allow for testing of the changes

Reviewers: jerryzh168

Subscribers: jerryzh168, supriyar

Test plan: In pytorch main dir, execute
```
python test_quantization.py TestPostTrainingStatic.test_quantized_embedding
```
to run the series of tests, including the newly added test_embedding_4bit
function

Tasks: T106931792

Tags

[ghstack-poisoned]

dzdang committed:
Update on "[Quant][Eager] Added 4 bit support for eager mode quantization flow"


Summary: Minor modifications were made to support 4 bit embedding quantized module in eager mode quantization flow and to allow for testing of the changes

Reviewers: jerryzh168

Subscribers: jerryzh168, supriyar

Test plan: In pytorch main dir, execute
```
python test_quantization.py TestPostTrainingStatic.test_quantized_embedding
```
to run the series of tests, including the newly added test_embedding_4bit
function

Tasks: T106931792

Tags

[ghstack-poisoned]

dzdang commented:
@dzdang has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33152675).
